plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java

```diff
@@ -176,6 +187,94 @@ public void testEnforcedCooldownPeriod() throws IOException {
+
+        final String snapshot = "snapshot";
+        assertSuccessfulSnapshot(client().admin().cluster().prepareCreateSnapshot(repository, snapshot)
+            .setWaitForCompletion(true).setIndices(index));
+
+        assertAcked(client().admin().indices().prepareDelete(index));
+
+        assertSuccessfulRestore(client().admin().cluster().prepareRestoreSnapshot(repository, snapshot).setWaitForCompletion(true));
+        ensureGreen(index);
+        assertHitCount(client().prepareSearch(index).setSize(0).setTrackTotalHits(true).get(), nbDocs);
+
+        assertAcked(client().admin().cluster().prepareDeleteSnapshot(repository, snapshot).get());
+
+        final RepositoryStats repositoryStats = StreamSupport.stream(
+            internalCluster().getInstances(RepositoriesService.class).spliterator(), false)
+            .map(repositoriesService -> {
+                try {
+                    return repositoriesService.repository(repository);
+                } catch (RepositoryMissingException e) {
+                    return null;
+                }
+            })
+            .filter(r -> r != null)
+            .map(r -> r.stats())
+            .reduce((s1, s2) -> s1.combine(s2))
+            .get();
+        final long sdkGetCalls = repositoryStats.requestCounts.get("GET");
+        final long sdkListCalls = repositoryStats.requestCounts.get("LIST");
+
+        final long getCalls = handlers.values().stream()
+            .mapToLong(h -> {
+                long count = 0;
+                while (h instanceof DelegatingHttpHandler) {
+                    if (h instanceof S3ErroneousHttpHandler) {
+                        count += ((S3ErroneousHttpHandler) h).getCalls.get();
+                    }
+                    h = ((DelegatingHttpHandler) h).getDelegate();
+                }
+                if (h instanceof S3HttpHandler) {
+                    return count + ((S3HttpHandler) h).getCalls.get();
+                } else {
+                    assert false;
+                    return count;
+                }
+            })
+            .sum();
+        final long listCalls = handlers.values().stream()
+            .mapToLong(h -> {
+                long count = 0;
+                while (h instanceof DelegatingHttpHandler) {
+                    if (h instanceof S3ErroneousHttpHandler) {
+                        count += ((S3ErroneousHttpHandler) h).listCalls.get();
+                    }
+                    h = ((DelegatingHttpHandler) h).getDelegate();
+                }
+                if (h instanceof S3HttpHandler) {
+                    return count + ((S3HttpHandler) h).listCalls.get();
+                } else {
+                    assert false;
+                    return count;
+                }
+            })
+            .sum();
+
+        logger.info("SDK sent {} GET calls and handler measured {} GET calls", sdkGetCalls, getCalls);
```

Comment:
Can you remove the log message and add them to the assertEquals() methods instead?